Library/Homebrew/services/cli.rb

```diff
@@ -196,18 +196,27 @@ def self.stop(targets, verbose: false, no_wait: false, max_wait: 0, keep: false)
               System::Systemctl.quiet_run(*systemctl_args, "disable", "--now", service.service_name)
             end
           elsif System.launchctl?
-            quiet_system System.launchctl, "bootout", "#{System.domain_target}/#{service.service_name}"
-            unless no_wait
-              time_slept = 0
-              sleep_time = 1
-              max_wait = T.must(max_wait)
-              while ($CHILD_STATUS.to_i == 9216 || service.loaded?) && (max_wait.zero? || time_slept < max_wait)
-                sleep(sleep_time)
-                time_slept += sleep_time
-                quiet_system System.launchctl, "bootout", "#{System.domain_target}/#{service.service_name}"
+            domain_targets = System.candidate_domain_targets
+            dont_wait_statuses = [3, 125] # 3 = no such process, 125 = invalid domain
+            while service.loaded? && !domain_targets.empty?
```

Comment:
I sort of hate this `while` loop (and am 👎🏻 on adding another) given it, in bad states, can (and does) hang indefinitely. A for loop with a maximum number of iterations feels nicer if possible?

Feels like e.g. 30-60 seconds: if a service hasn't stopped: it's never going to.

I also think it's worth attempting the `stop` regardless, even if the `bootout` never completes, particularly given we have a few different targets to try now.

I also wonder whether failing in this case might be a (very rare for Homebrew) case where we recommend `sudo brew services stop`, assuming that will attempt to stop for any/all users.

Reply:
thanks for context! `domain_targets.each` would probably make that more obvious.

Reply:
> Feels like e.g. 30-60 seconds: if a service hasn't stopped: it's never going to.
> 
> I also think it's worth attempting the `stop` regardless, even if the `bootout` never completes, particularly given we have a few different targets to try now.

these also feel worth addressing here

Reply:
Does that mean changing the default value of `--max-wait`?

Reply:
@Bo98 Yes, I think that makes sense. I don't think an infinite default wait makes sense.